l10n_it_fatturapa_in/wizard/link_to_existing_invoice.py

```diff
@@ -10,10 +12,42 @@ class WizardLinkToInvoice(models.TransientModel):
     invoice_id = fields.Many2one(
         'account.invoice', string="Invoice", required=True)
 
+    def get_invoice_obj(self, fatturapa_attachment):
+        xml_string = fatturapa_attachment.get_xml_string()
+        return fatturapa_v_1_2.CreateFromDocument(xml_string)
+
     @api.multi
     def link(self):
         self.ensure_one()
         active_ids = self.env.context.get('active_ids')
         if len(active_ids) != 1:
             raise UserError(_("You can select only 1 XML file to link"))
         self.invoice_id.fatturapa_attachment_in_id = active_ids[0]
+        # extract pdf if attached
+        fatturapa_attachment_obj = self.env['fatturapa.attachment.in']
+        fatturapa_attachment_ids = self.env.context.get('active_ids', False)
+        for fatturapa_attachment_id in fatturapa_attachment_ids:
+            fatturapa_attachment = fatturapa_attachment_obj.browse(
+                fatturapa_attachment_id)
+            fatt = self.get_invoice_obj(fatturapa_attachment)
+            for FatturaBody in fatt.FatturaElettronicaBody:
+                # 2.5
```

Comment:
@sergiocorato potresti mettere questa parte, che è uguale in `wizard_import_fatturapa.py`, in un metodo a parte da chiamare da qui e da `wizard_import_fatturapa.py`?

Reply:
fatto